.github/workflows/daily-dev-bump.yaml

```diff
@@ -54,11 +60,43 @@ jobs:
         # Do the update
         dart tool/update_version.dart auto --type dev
 
-    - name: commit
+    - name: Create the PR
       run: |
+        BRANCH_NAME="auto-bump-$(date +%s)"
         # Stage the file, commit and push
-
+        git checkout -b "$BRANCH_NAME"
         git commit -a -m "$COMMIT_MESSAGE"
-        git push
+        git push -u origin "$BRANCH_NAME"
+
+        PR_URL=$(gh pr create --title "$COMMIT_MESSAGE" --body "RELEASE_NOTE_EXCEPTION=Automated Version Bump")
+
+        # Change github credentials back to the actions bot.
+        GH_TOKEN="$ORIGINAL_GH_TOKEN"
+        gh pr edit $PR_URL --add-label "autosubmit"
+
       env:
         COMMIT_MESSAGE: ${{ steps.version-bump.outputs.COMMIT_MESSAGE }}
+        GH_TOKEN: ${{ secrets.DEVTOOLS_WORKFLOW_BOT_TOKEN }}
+        ORIGINAL_GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  clean-up-branches:
+    # If a pr is closed on a workflow bot PR, then clean up workflow bot branches.
+    if: ${{ github.event_name == 'pull_request' && github.event.action == 'closed'  && github.event.pull_request.user.login == 'DartDevtoolWorkflowBot'}}
```

Comment:
is there a difference between 'closed' and 'merged' for the github.event.action?

Reply:
I believe 'closed' is triggered for both merging the pr and just closing it.
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges

So I believe it is an appropriate catch for when we want to be cleaning up the dangling branches :)